### PR TITLE
use query params to filter name and type on DME servers

### PR DESCRIPTION
**Pull request by mgbatchelor**
This approach is much faster, and reduces the amount of data that needs to be loaded, parsed, and then thrown away.

Documentation is located: https://www.dnsmadeeasy.com/pdf/API-Docv2.pdf

### CODE REVIEW

File: lib/dnsmadeeasy-rest-api.rb [hunk 1]
```diff
@@ -59,15 +59,15 @@ def records_for(domain_name)
   end
 
   def find(domain_name, name, type)
-    records = records_for(domain_name)
-    records['data'].detect { |r| r['name'] == name && r['type'] == type }
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
```
Comment (codealchemy) [inline lib/dnsmadeeasy-rest-api.rb]: We don't have `#find` implemented in [`nbuild`](https://github.com/3dna/nbuild/blob/master/lib/nation_builder/dns/dns_made_easy_adapter.rb), would that be a separate add after this?

File: lib/dnsmadeeasy-rest-api.rb [hunk 2]
```diff
@@ -59,15 +59,15 @@ def records_for(domain_name)
...
+
+    records['data'].first
   end
 
   def find_record_id(domain_name, name, type)
-    records = records_for(domain_name)
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
```
Comment (codealchemy) [inline lib/dnsmadeeasy-rest-api.rb]: Think it might be worth extracting this call since it's shared between the two here.

Comment (mgbatchelor) [reply lib/dnsmadeeasy-rest-api.rb]: updated in https://github.com/3dna/dnsmadeeasy-rest-api/pull/2/commits/95f58fecb29d6c25633c7b17bad224edfe14f8d8 to make it also work for records_for

**Review by codealchemy — APPROVED** [review]
💯

**Pull request merged**